src/components/screens/on-call-calendar-screen.tsx

```diff
@@ -0,0 +1,259 @@
+import OnCallListView from "../oncall-calendars/oncall-list-view";
+import type { OnCallPaid } from "src/generated/homeLambdasClient";
+
+/**
+ * On call calendar screen component
+ */
+const OnCallCalendarScreen = () => {
+  /**
+   * Validates if local storage item is of correct type
+   * @returns boolean value from local storage if validated
+   */
+  const validateJSONString = () => {
+    const item = localStorage.getItem("calendarViewAsDefault");
+    if (item) return JSON.parse(item);
+    return true;
+  };
+
+  const { onCallApi } = useLambdasApi();
+  const [onCallData, setOnCallData] = useAtom(onCallAtom);
+  const [open, setOpen] = useState(false);
+  const [isCalendarView, setIsCalendarView] = useState(validateJSONString());
+  const [selectedDate, setSelectedDate] = useState<DateTime>(DateTime.now());
+  const [onCallPerson, setOnCallPerson] = useState("");
+  const [selectedOnCallEntry, setSelectedOnCallEntry] = useState<OnCallCalendarEntry>();
+  const setError = useSetAtom(errorAtom);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    getOnCallData(selectedDate.year);
+  }, [selectedDate.year]);
+
+  // biome-ignore lint/correctness/useExhaustiveDependencies: dependency is necessary
+  useEffect(() => {
+    getCurrentOnCallPerson();
+  }, [onCallData]);
+
+  /**
+   * Fetches on call data, such as personnel and weeks
+   * @param year year of entries to search
+   */
+  const getOnCallData = async (year: number) => {
+    setIsLoading(true);
+    try {
+      const fetchedData = await onCallApi.listOnCallData({ year: year.toString() });
+      setOnCallData(fetchedData);
+    } catch (error) {
+      setError(`${strings.oncall.fetchFailed}, ${error}`);
+    }
+    setIsLoading(false);
+  };
+
+  /**
+   * Finds the current on call employee and sets them in a state
+   */
+  const getCurrentOnCallPerson = () => {
+    if (selectedDate.year === DateTime.now().year) {
+      const currentWeek = DateTime.now().weekNumber;
+      const currentOnCallPerson = onCallData.find((item) => Number.parseInt(item.week) === currentWeek)?.person;
+      if (currentOnCallPerson) setOnCallPerson(currentOnCallPerson);
+    }
+  };
+
+  /**
+   * Generates props for Badge component to be displayed in DateCalendar
+   * @returns Array containing props for Badge component
+   */
+  const generateOnCallWeeks = () => {
+    const onCallWeeks: OnCallCalendarEntry[] = [];
+    for (const item of onCallData) {
+      const weeks = DateTime.fromObject({ weekNumber: Number.parseInt(item.week), weekYear: selectedDate.year });
+
+      for (let i = 0; i < 7; i++) {
+        onCallWeeks.push({
+          date: weeks.plus({ days: i }).toISODate(),
+          person: item.person,
+          paid: item.paid,
+          badgeColor: stringToColor(item.person)
+        });
+      }
+    }
+    return onCallWeeks;
+  };
+
+  /**
+   * Updates the selected paid status
+   * @param year
```

Comment:
It's good to have a bit for info in this doc e.g. 
```suggestion
   * @param year number
```

Please update other param docs that you see are missing the type of the param